lib/srv/app/session.go

```diff
@@ -171,12 +172,25 @@ func (s *Server) withJWTTokenForwarder(ctx context.Context, sess *sessionChunk,
 		return trace.Wrap(err)
 	}
 
+	// Duplicate the default behavior for the header rewriter in oxy, as there
+	// is no getters/setters for this in the Forwarder so that we can grab it
+	// and re-set it there.
+	h, err := os.Hostname()
+	if err != nil {
+		h = "localhost"
+	}
+	delegate := &forward.HeaderRewriter{TrustForwardHeader: true, Hostname: h}
+
+	hr := &headerRewriter{
+		delegate: delegate,
+	}
 	sess.fwd, err = forward.New(
 		forward.FlushInterval(100*time.Millisecond),
 		forward.RoundTripper(transport),
 		forward.Logger(logrus.StandardLogger()),
 		forward.WebsocketRewriter(transport.ws),
 		forward.WebsocketDial(transport.ws.dialer),
+		forward.Rewriter(hr),
```

Comment:
Couple questions here:

- Is this the only place we need to set our custom rewriter? We also create forwarder in web/apps.
- What about websocket requests rewriter?

Reply:
Let me take a look at those. Thanks for letting me know about those.

Reply:
I've addressed this, but there's some more explanation in the PR description.